core/services/ocr2/delegate.go

```diff
@@ -986,6 +992,21 @@ func (d *Delegate) newServicesOCR2Functions(
 		ReportingPluginFactory:       nil, // To be set by NewFunctionsServices
 	}
 
+	s4OracleArgs := libocr2.OCR2OracleArgs{
+		BinaryNetworkEndpointFactory: d.peerWrapper.Peer2,
+		V2Bootstrappers:              bootstrapPeers,
+		ContractTransmitter:          functionsProvider.ContractTransmitter(),
+		ContractConfigTracker:        functionsProvider.ContractConfigTracker(),
```

Comment:
This is not going to work. We need a separate instance of functionsProvider for each plugin with corresponding FunctionsPluginType set.
I wonder if we could do some minor refactor to avoid too much copy-paste for 3 different plugins...

Reply:
Changed.